test/test_python.py

```diff
@@ -297,9 +297,14 @@ def test_run_core_python_test(python_test, selenium, request):
     try:
         selenium.run(
             "from test.libregrtest import main\n"
-            "main(['{}'], verbose=True, verbose3=True)".format(name))
+            "try:\n"
+            "    main(['{}'], verbose=True, verbose3=True)\n"
+            "except SystemExit as e:\n"
+            "    if e.code != 0:\n"
+            "        raise RuntimeError(f'Failed with code: {{e.code}}')\n".format(name))
```

Comment:
Due to changes in Python itself, `SystemExit` now bubbles all the way up to call the system call `abort`.  That's probably an improvement, as it's more like "real" Python.  However, it does mean with have to explicitly catch `SystemExit` now.